src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -62,13 +69,24 @@
         this.transaction = trnsct;
     }
 
-    // @todo #15:30min Implement id() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
     @SuppressWarnings("PMD.ShortMethodName")
-    public long id() {
-        throw new UnsupportedOperationException("id() not yet implemented");
+    public long id() throws IOException {
+        final String ident = new IoCheckedScalar<String>(
+            () -> new ItemAt<>(
```

Comment:
@carlosmiranda  you should be able to directly pass the object which implements `Scalar` here, no need for a lambda

Reply:
@victornoel see https://github.com/zold-io/java-api/pull/45#discussion_r205359262